### PR TITLE
[6088] Fix string range filter with wrong statistics

### PR DESCRIPTION
**Pull request by loukey-lj**
https://github.com/facebookincubator/velox/issues/6088

### CODE REVIEW

**Comment by yingsu00** [comment]
@nmahadevuni Will you be able to take a look?

File: velox/dwio/common/ScanSpec.cpp [hunk 1]
```diff
@@ -263,22 +263,28 @@ bool testStringFilter(
...
-    const auto& min = stringStats->getMinimum().value();
-    const auto& max = stringStats->getMaximum().value();
+ auto min_optional = stringStats->getMinimum();
+ auto max_optional = stringStats->getMaximum();
+
+  if (min_optional.has_value() && max_optional.has_value()) {
+    const auto& min = min_optional.value();
+    const auto& max = max_optional.value();
+    if(min > max) {
+      return true;
```
Comment (nmahadevuni) [inline velox/dwio/common/ScanSpec.cpp:273]: Since we are working around corrupt statistics, can we log a warning?

**Comment by nmahadevuni** [comment]
@loukey-lj Please sign up for CLA at https://code.facebook.com/cla. Also use scripts/check.py to correct the code format which is failing.

**Review by yingsu00 — COMMENTED** [review]
@loukey-lj Are you still working this?

**Comment by loukey-lj** [comment]
@yingsu00 Sorry, I haven't dealt with this matter anymore. Let interested parties follow up on this issue.

**Pull request closed without merging**